### PR TITLE
[SM-670] Add permission context to project lists.

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The purpose of this PR is to add permission context to project lists and cleanup the access checks provided in the project repository.

This is to enable hiding UI elements based on current user permission.

## Code changes

- bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/CreateAccessPoliciesCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/DeleteAccessPolicyCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/UpdateAccessPolicyCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Projects/DeleteProjectCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Projects/UpdateProjectCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/CreateSecretCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/DeleteSecretCommand.cs
bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/UpdateSecretCommand.cs
src/Api/SecretsManager/Controllers/AccessPoliciesController.cs

Swap to just using `Task<(bool Read, bool Write)> AccessToProjectAsync(Guid id, Guid userId, AccessClientType accessType);`.

On some of these, we are specifically preventing service account access.

- bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/ProjectRepository.cs

Swap to using `ProjectPermissionDetails` to provide the current user's permissions.
Swap to using `Task<(bool Read, bool Write)> AccessToProjectAsync(Guid id, Guid userId, AccessClientType accessType);` and remove old access check methods. 

- bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/CreateAccessPoliciesCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/DeleteAccessPolicyCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/UpdateAccessPolicyCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/Projects/DeleteProjectCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/Projects/UpdateProjectCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/Secrets/CreateSecretCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/Secrets/DeleteSecretCommandTests.cs
bitwarden_license/test/Commercial.Core.Test/SecretsManager/Secrets/UpdateSecretCommandTests.cs
test/Api.Test/SecretsManager/Controllers/AccessPoliciesControllerTests.cs
test/Api.Test/SecretsManager/Controllers/SecretsControllerTests.cs

Update unit tests to use `Task<(bool Read, bool Write)> AccessToProjectAsync(Guid id, Guid userId, AccessClientType accessType);`.

- src/Api/SecretsManager/Controllers/ProjectsController.cs
src/Api/SecretsManager/Models/Response/ProjectResponseModel.cs
src/Core/SecretsManager/Models/Data/ProjectPermissionDetails.cs

Updating request model to return current user's permissions.

- src/Core/SecretsManager/Repositories/IProjectRepository.cs

Remove old access check interfaces.

- test/Api.IntegrationTest/SecretsManager/Controllers/ProjectsControllerTests.cs
test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs
test/Api.Test/SecretsManager/Controllers/ProjectsControllerTests.cs

Updating unit tests to new models.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/SecretsManager/Models/Data/ProjectPermissionDetails.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@ namespace Bit.Core.SecretsManager.Models.Data;
 
 public class ProjectPermissionDetails : Project
 {
+    public Project Project;
```
Comment (coltonhurst) [inline src/Core/SecretsManager/Models/Data/ProjectPermissionDetails.cs:7]: `ProjectPermessionDetails` is already a `Project`. Why are we adding this?

**Review by coltonhurst — CHANGES_REQUESTED** [review]
Looking good, one main question so far.

File: src/Core/SecretsManager/Models/Data/ProjectPermissionDetails.cs (hunk 1 above)
Comment (Thomas-Avery) [reply src/Core/SecretsManager/Models/Data/ProjectPermissionDetails.cs:7]: Good catch, the intention was to match `SecretPermissionDetails` with this update for consistency. 

I forgot to remove the inheritance when altering the model.

Should be fixed in https://github.com/bitwarden/server/pull/2822/commits/1d99670bd9b5555254a11b719597a7833f5270f2

**Review by coltonhurst — APPROVED** [review]
Looks great, thank you!

**Review by coltonhurst — APPROVED** [review]

**Review by coltonhurst — APPROVED** [review]

**Pull request merged**